services/wallet/router/router_v2.go

```diff
@@ -326,6 +328,34 @@ func validateInputData(input *RouteInputParams) error {
 		}
 	}
 
+	if input.SendType == Swap {
```

Comment:
The function `validateInputData` is kind of the perfect function to me to be unit tested because it's pure and it has so many conditions that I doubt we will ever cover them all with slow integration tests, more likely that integration tests will focus on a happy path and only a few sad paths. Therefore, there's a high chance that we'll end up with uncovered paths when the router is sort of finished.

Reply:
Exactly, that's why we have it as a separate function, that's what @Samyoul did in this PR https://github.com/status-im/status-go/pull/5323 so we already have that, just needs to be merged.

Reply:
Right, missed that. Thanks!